### PR TITLE
feat: Metadata as json

### PR DESCRIPTION
**Pull request by ron-gal**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/langchain-google-bigtable-python/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: src/langchain_google_bigtable/document_loader.py [hunk 1]
```diff
@@ -110,6 +121,9 @@ def __init__(
         self.content_column_family = content_column_family
         self.content_column_name = content_column_name
         self.metadata_mappings = metadata_mappings
+        self.metadata_as_json_column_family = metadata_as_json_column_family
+        self.metadata_as_json_column_name = metadata_as_json_column_name
+        self.metadata_as_json_encoding = metadata_as_json_encoding
```
Comment (kurtisvg) [inline src/langchain_google_bigtable/document_loader.py:126]: Should we throw an error if one of these is specified but not the other? 

And probably verify they exist?

File: src/langchain_google_bigtable/document_loader.py [hunk 2]
```diff
@@ -132,6 +146,20 @@ def lazy_load(
...
+                and self.metadata_as_json_column_name.encode()
+                in row.cells[self.metadata_as_json_column_family]
+            ):
+                metadata.update(
+                    json.loads(
+                        row.cells[self.metadata_as_json_column_family][
+                            self.metadata_as_json_column_name.encode()
+                        ][0].value.decode(self.metadata_as_json_encoding.value)
+                    )
+                )
```
Comment (kurtisvg) [inline src/langchain_google_bigtable/document_loader.py:162]: I'm inclined to think we should load this as metadata first, and overwrite the columns into it

Comment (kurtisvg) [inline src/langchain_google_bigtable/document_loader.py:162]: nit: this is like half the lines and twice as readable
```suggestion
                col_family, col_name = self.metadata_as_json_column_family, self.metadata_as_json_column_name
                r =  row.cells[col_family][col_name.encode()][0]
                j = json.loads(r.value.decode(self.metadata_as_json_encoding.value))
                metadata.update(json.loads(r.value.decode(self.metadata_as_json_encoding.value)))
```

File: src/langchain_google_bigtable/document_loader.py [hunk 3]
```diff
@@ -223,10 +264,25 @@ def add_documents(self, docs: List[Document]):
...
+            unmapped_metadata = dict(doc.metadata)
+            unmapped_metadata.pop(ID_METADATA_KEY, None)
             for mapping in self.metadata_mappings:
                 if mapping.metadata_key in doc.metadata:
                     value = self._encode(doc.metadata[mapping.metadata_key], mapping)
                     row.set_cell(mapping.column_family, mapping.column_name, value)
+                    unmapped_metadata.pop(mapping.metadata_key)
+            if (
+                self.metadata_as_json_column_family is not None
+                and self.metadata_as_json_column_name is not None
```
Comment (kurtisvg) [inline src/langchain_google_bigtable/document_loader.py:276]: I'm concerned about one of these being none and the other just being ignored

**Review by kurtisvg — CHANGES_REQUESTED** [review]

File: src/langchain_google_bigtable/document_loader.py (hunk 1 above)
Comment (ron-gal) [reply src/langchain_google_bigtable/document_loader.py:126]: We don't verify any of the mappings exist. If we find them - we populate them.
I could check, similarly to how I do it in the ChatMessageHistory. Should I?

Comment (ron-gal) [reply src/langchain_google_bigtable/document_loader.py:126]: Done

File: src/langchain_google_bigtable/document_loader.py (hunk 3 above)
Comment (ron-gal) [reply src/langchain_google_bigtable/document_loader.py:276]: Fixed. If one is None and the other is not - we throw in the ctor.

File: src/langchain_google_bigtable/document_loader.py (hunk 2 above)
Comment (ron-gal) [reply src/langchain_google_bigtable/document_loader.py:162]: Done

Comment (ron-gal) [reply src/langchain_google_bigtable/document_loader.py:162]: Done

File: src/langchain_google_bigtable/document_loader.py [hunk 4]
```diff
@@ -121,6 +160,22 @@ def lazy_load(
         rows = self.client.read_rows(row_set=self.row_set, filter_=self.filter)
         for row in rows:
             metadata = {ID_METADATA_KEY: row.row_key.decode()}
+            col_family, col_name = (
+                self.metadata_as_json_column_family,
+                self.metadata_as_json_column_name,
+            )
```
Comment (kurtisvg) [inline src/langchain_google_bigtable/document_loader.py:166]:
```suggestion
            col_family = self.metadata_as_json_column_family
            col_name = self.metadata_as_json_column_name
```

**Review by kurtisvg — APPROVED** [review]

**Pull request merged**